### DISCUSSION TITLE
How to read any files?

### DISCUSSION THREAD

**Discussion by ngdangtu-vn** [tauri-apps/tauri#7850] [Q&A]
I plan to build a text editor, and I can not ask users copy their files to a scope directory for my app to load. Is there anyway to open file(s) from anywhere in PC with user permission?

I prefer Rust way for performance.

**Comment by FabianLars** [comment]
If you want to read the files in rust then you don't need to worry about the scope anyway. It will only be applied to the javascript side. That said, if you use tauri's [file dialog](https://docs.rs/tauri/latest/tauri/api/dialog/blocking/struct.FileDialogBuilder.html) the scope will be expanded automatically. You can also expand the scope dynamically yourself: https://docs.rs/tauri/latest/tauri/trait.Manager.html#method.fs_scope (that really needs a doc example...)
